ulauncher/api/server/ExtensionSearchMode.py

```diff
@@ -55,7 +55,12 @@ def get_searchable_items(self):
                 if not pref['value']:
                     continue
 
-                icon = c.manifest.load_icon(ExtensionKeywordResultItem.get_icon_size())
+                if pref['icon'] is not None:
+                    icon = c.manifest.load_icon_from_path(pref['icon'], ExtensionKeywordResultItem.get_icon_size())
+                else:
+                    icon = c.manifest.load_icon(
+                        ExtensionKeywordResultItem.get_icon_size())
```

Comment:
Might move this back up to the same line as the assignment.

Reply:
Agreed. These two methods are nearly the same, and the old one is only used in a few places, so I think it's better to refactor it to work with an optional path if needed. Although I'm a bit confused because it seems all you're doing is passing `pref['icon']`. I though that's what the original method defaults to?